seedlet/inc/wpcom-colors.php

```diff
@@ -539,4 +539,107 @@ function seedlet_custom_colors_extra_css() {
+
+		add_theme_support(
+			'editor-color-palette',
+			array(
+				array(
+					'name'  => __( 'Primary', 'seedlet' ),
+					'slug'  => 'primary',
+					'color' => $primary
+				),
+				array(
+					'name'  => __( 'Secondary', 'seedlet' ),
+					'slug'  => 'secondary',
+					'color' => $secondary
+				),
+				array(
+					'name'  => __( 'Foreground', 'seedlet' ),
+					'slug'  => 'foreground',
+					'color' => $foreground
+				),
+				array(
+					'name'  => __( 'Tertiary', 'seedlet' ),
+					'slug'  => 'tertiary',
+					'color' => $tertiary
+				),
+				array(
+					'name'  => __( 'Background', 'seedlet' ),
+					'slug'  => 'background',
+					'color' => $background
+				),
+			)
+		);
+
+		$gradient_color_a = $secondary;
+		$gradient_color_b = $tertiary;
+
+		add_theme_support(
+			'editor-gradient-presets',
+			array(
+				array(
+					'name'     => __( 'Diagonal', 'seedlet' ),
+					'gradient' => 'linear-gradient(to bottom right, ' . $gradient_color_a . ' 49.9%, ' . $gradient_color_b  . ' 50%)',
+					'slug'     => 'hard-diagonal',
+				),
+				array(
+					'name'     => __( 'Diagonal inverted', 'seedlet' ),
+					'gradient' => 'linear-gradient(to top left, ' . $gradient_color_a . ' 49.9%, ' . $gradient_color_b . ' 50%)',
+					'slug'     => 'hard-diagonal-inverted',
+				),
+				array(
+					'name'     => __( 'Horizontal', 'seedlet' ),
+					'gradient' => 'linear-gradient(to bottom, ' . $gradient_color_a . ' 50%, ' . $gradient_color_b . ' 50%)',
+					'slug'     => 'hard-horizontal',
+				),
+				array(
+					'name'     => __( 'Horizontal inverted', 'seedlet' ),
+					'gradient' => 'linear-gradient(to top, ' . $gradient_color_a . ' 50%, ' . $gradient_color_b . ' 50%)',
+					'slug'     => 'hard-horizontal-inverted',
+				),
+				array(
+					'name'     => __( 'Diagonal gradient', 'seedlet' ),
+					'gradient' => 'linear-gradient(to bottom right, ' . $gradient_color_a . ', ' . $gradient_color_b . ')',
+					'slug'     => 'diagonal',
+				),
+				array(
+					'name'     => __( 'Diagonal inverted gradient', 'seedlet' ),
+					'gradient' => 'linear-gradient(to top left, ' . $gradient_color_a . ', ' . $gradient_color_b . ')',
+					'slug'     => 'diagonal-inverted',
+				),
+				array(
+					'name'     => __( 'Horizontal gradient', 'seedlet' ),
+					'gradient' => 'linear-gradient(to bottom, ' . $gradient_color_a . ', ' . $gradient_color_b . ')',
+					'slug'     => 'horizontal',
+				),
+				array(
+					'name'     => __( 'Horizontal inverted gradient', 'seedlet' ),
+					'gradient' => 'linear-gradient(to top, ' . $gradient_color_a . ', ' . $gradient_color_b . ')',
+					'slug'     => 'horizontal-inverted',
+				),
+				array(
+					'name'     => __( 'Stripe', 'seedlet' ),
+					'gradient' => 'linear-gradient(to bottom, transparent 20%, ' . $gradient_color_a . ' 20%, ' . $gradient_color_a . ' 80%, transparent 80%)',
+					'slug'     => 'stripe',
+				),
+			)
+		);
+	}
+}
+add_action( 'after_setup_theme', 'seedlet_replace_editor_palette' );
```

Comment:
This approach isn't working in my testing, but I wanted to push this to see if anyone had any ideas.

Reply:
I think this needs to go in `wpcom.php`, not `wpcom-colors.php`. The latter file is normally used just for the color annotations themselves, so I think it may be pulled in too late to make this work.

Reply:
Ah good call! Moving this code to `wpcom.php` loads the editor palette correctly.